rest-service/manager_rest/rest/resources_v3_1/executions.py

```diff
@@ -13,12 +13,122 @@
+
+from flask import request
+from flask_restful.inputs import boolean
+from flask_restful.reqparse import Argument
+
+from cloudify._compat import text_type
+from cloudify.models_states import ExecutionState
 
 from manager_rest.security import SecuredResource
 from manager_rest.security.authorization import authorize
+from manager_rest.rest import resources_v2, rest_decorators
 from manager_rest.resource_manager import get_resource_manager
-from manager_rest.storage import (models,
-                                  get_storage_manager)
+from manager_rest.rest.rest_utils import (get_args_and_verify_arguments,
+                                          get_json_and_verify_params)
+from manager_rest.storage import models,  get_storage_manager, ListResult
+
+
+class Executions(resources_v2.Executions):
+    @authorize('execution_delete')
+    @rest_decorators.marshal_with(models.Execution)
+    @rest_decorators.paginate
+    def delete(self, pagination=None):
+        request_dict = get_json_and_verify_params({
+            'keep_last': {'optional': True, 'type': int},
+            'keep_days': {'optional': True, 'type': int},
+            'keep_since': {'optional': True}
+        })
+        keep_since = None
+        if 'keep_since' in request_dict:
+            # this will raise ValueError if `keep_since` is not in date format
+            keep_since = \
+                datetime.strptime(request_dict['keep_since'], "%Y-%m-%d")
+        if ('keep_last' in request_dict and request_dict['keep_last'] <= 0) or\
+                ('keep_days' in request_dict and
+                 request_dict['keep_days'] <= 0):
+            raise ValueError('Must specify an integer greater than 0')
+
+        sm = get_storage_manager()
+        args = get_args_and_verify_arguments(
+            [Argument('created_by', type=text_type, required=False),
+             Argument('status', type=text_type, required=False),
+             Argument('all_tenants', type=boolean, default=False)]
+        )
+        filters = {}
+        if args['created_by']:
+            filters['created_by'] = args['created_by']
+        if not args['all_tenants']:
+            filters['tenant_name'] = request.headers.get('Tenant')
+        if args['status']:
+            if args['status'] not in ExecutionState.END_STATES:
+                raise ValueError(
+                    'Can\'t filter by execution status `{0}`. '
+                    'Allowed statuses are: {1}'.format(
+                        args['status'], ExecutionState.END_STATES)
+                )
+            filters['status'] = args['status']
+        else:
+            filters['status'] = ExecutionState.END_STATES
+
+        executions = sm.list(models.Execution,
+                             filters=filters,
+                             all_tenants=args['all_tenants'])
+        dep_creation_execs = {}
+        for execution in executions:
+            if execution.workflow_id == 'create_deployment_environment' and \
```

Comment:
i'm not really sure what does this +1 -1 logic do, but i think perhaps we dont need to allow deleting create-dep-env executions at all